src/main/java/core/basesyntax/service/impl/operation/BalanceOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.Fruit;
+
+public class BalanceOperation implements OperationHandler {
+    private final FruitStorageDao fruitStorageDao;
+
+    public BalanceOperation(FruitStorageDao fruitStorageDao) {
+        this.fruitStorageDao = fruitStorageDao;
+    }
+
+    @Override
+    public boolean doOperation(Fruit fruit, int quantity) {
+        if (fruit == null) {
+            throw new RuntimeException("Fruit can't be null");
+        }
+        if (quantity < MIN_FRUIT_QUANTITY) {
+            throw new RuntimeException("Quantity " + quantity + " can't be negative");
```

Comment:
The `MIN_FRUIT_QUANTITY` variable is not defined in this class, which will cause a compilation error. You need to define this constant in this class or use a different way to validate the quantity. Also, the condition should check if the quantity is less than 0, because a balance cannot be negative.